.github/workflows/update-ss3-models.yml

```diff
@@ -0,0 +1,95 @@
+name: update-user-example-models
+on: workflow_dispatch:
+
+jobs:
+  update-user-example-models:
+    runs-on: ubuntu-latest
+    env:
+      R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+
+    steps:
+
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      
+      - name: install libcurl
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libcurl4-openssl-dev
+
+      - name: setup R  
+        uses: r-lib/actions/setup-r@v2
+      
+      - name: install remotes
+        run: Rscript -e 'install.packages("remotes")'
+
+      - name: install r4ss
+        run: Rscript -e 'remotes::install_github("r4ss/r4ss")'
+      
+      - name: install SSutils
+        run: Rscript -e 'remotes::install_github("r4ss/SSutils")'
+        
+      - name: Get the latest SS3 executable
+        run:  wget -O ss https://github.com/nmfs-stock-synthesis/stock-synthesis/releases/latest/download/ss_linux
+          
+      - name: move ss to path
+        run: |
+          echo "ss" >> $GITHUB_PATH
+     
+      - name: run models
+        run: |
+          # first part: update the reference files.
+          # run models first
+          dirs_test <- list.dirs(recursive = FALSE)
+          mod_dir_name <- dirs_test[grepl("model", dirs_test)]
+          mod_dir_name <- gsub(".*\\./", "", mod_dir_name)
+          mod_dirs <- list.dirs(dirs_test[grepl("model", dirs_test)], recursive = FALSE)
+
+          r4ss::populate_multiple_folders(outerdir.old = mod_dir_name,
+                                          outerdir.new = mod_dir_name,
+                                          exe.file = NULL, verbose = FALSE)
+          # use the parallel version to speed up model runs 
+          model_runs <- SSutils::run_SS_models_parallel(dirvec = mod_dirs,
+                                          model = "ss", exe_in_path = TRUE, 
+                                          parallel = TRUE, verbose = TRUE)
+
+          # test that all the models ran successfully
+          if (!all(model_runs$results == "ran model")) {
+            stop("Not all models ran successfully.") # unknown run status is possible
```

Comment:
interesting. Do you know why the run status could be unknown instead of failing?

Reply:
This was legacy from Kelli's R code that she wrote and to update model files. I thought it might be good to keep it in there just in case.